napari/_qt/_tests/test_qt_viewer.py

```diff
@@ -254,11 +254,28 @@ def test_screenshot(make_napari_viewer):
 
     # Take screenshot
     with pytest.warns(FutureWarning):
-        screenshot = viewer.window.qt_viewer.screenshot(flash=False)
+        viewer.window.qt_viewer.screenshot(flash=False)
     screenshot = viewer.window.screenshot(flash=False, canvas_only=True)
     assert screenshot.ndim == 3
 
 
+def test_screenshot_fit_to_data(make_napari_viewer):
+    viewer = make_napari_viewer()
+
+    np.random.seed(0)
+    # Add image
+    data = np.random.randint(150, 250, size=(250, 250))
+    viewer.add_image(data)
+    img = viewer.screenshot(flash=False, fit_to_data=True)
+    assert img.shape == (250, 250, 4)
+    assert np.all(img != np.array([0, 0, 0, 1]))
+
```

Comment:
This only fails when running the test itself. If you use the same code in a python script it does not fail. For some reason, there is no response to the canvas resize event when running inside a test.